projects/storefrontlib/src/cms-components/myaccount/my-interests/my-interests.component.spec.ts

```diff
@@ -86,180 +88,137 @@ const mockedInterests: ProductInterestList = {
-        { provide: AuthService, useClass: MockAuthService },
+        { provide: UserService, useValue: userService },
+        { provide: AuthService, useValue: authService },
+        { provide: OccConfig, useValue: MockOccModuleConfig },
+        { provide: LayoutConfig, useValue: MockLayoutConfig },
       ],
-      declarations: [MyInterestsComponent, MockTranslateUrlPipe],
+      declarations: [MyInterestsComponent, MockUrlPipe, MockMediaComponent],
     }).compileComponents();
-
-    userService = TestBed.get(UserService);
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(MyInterestsComponent);
     component = fixture.componentInstance;
     el = fixture.debugElement;
-    fixture.detectChanges();
-  });
 
-  it('should create', () => {
-    expect(component).toBeTruthy();
+    authService.getUserToken.and.returnValue(of(userToken));
+    userService.getProdutInterests.and.returnValue(of(emptyInterests));
+    userService.getProdutInterestsLoaded.and.returnValue(of(true));
+    userService.loadProductInterests.and.stub();
   });
 
-  it('should display title with empty interests', () => {
-    const emptyInterests: ProductInterestList = {
-      sorts: [{ code: 'name', asc: true }],
-      pagination: {},
-    };
-    spyOn(userService, 'getProdutInterests').and.returnValue(
-      of(emptyInterests)
-    );
-
-    component.ngOnInit();
+  it('should create', () => {
     fixture.detectChanges();
-
-    expect(el.query(By.css('.cx-product-interests-header'))).toBeTruthy();
+    expect(component).toBeTruthy();
   });
 
-  it('should display title with interests', () => {
-    const emptyInterests: ProductInterestList = {
-      sorts: [{ code: 'name', asc: true }],
-      pagination: {},
-      results: [],
-    };
-    spyOn(userService, 'getProdutInterests').and.returnValue(
-      of(emptyInterests)
-    );
-
-    component.ngOnInit();
+  it('should display message when no interest', () => {
     fixture.detectChanges();
-
     expect(el.query(By.css('.cx-product-interests-header'))).toBeTruthy();
   });
 
-  it('should display no interests message', () => {
-    const interests: ProductInterestList = {
-      sorts: [{ code: 'name', asc: true }],
-      pagination: {
-        count: 0,
-        page: 10,
-        totalCount: 0,
-        totalPages: 0,
-      },
-      results: [],
-    };
-    spyOn(userService, 'getProdutInterests').and.returnValue(of(interests));
-
-    component.ngOnInit();
+  it('should show interests list', () => {
+    userService.getProdutInterests.and.returnValue(of(mockedInterests));
+    userService.getProdutInterestsLoaded.and.returnValue(of(true));
     fixture.detectChanges();
 
-    expect(el.query(By.css('.cx-product-interests-no-interests'))).toBeTruthy();
+    expect(el.queryAll(By.css('cx-sorting')).length).toEqual(2);
+    expect(el.queryAll(By.css('cx-pagination')).length).toEqual(2);
+    expect(el.queryAll(By.css('tr')).length).toEqual(2);
```

Comment:
we also should verify the table content element is here, no need verify the content is right or not.
and whether the hyperlink is here should be verified

Reply:
and delete button should be verified too